public/utils/datasource-utils.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ *   Copyright OpenSearch Contributors
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { DataSourceOption } from 'src/plugins/data_source_management/public/components/data_source_menu/types';
+
+export function createDataSourceQuery(dataSourceId: string) {
```

Comment:
This function doesn't do much. What is the advantage of having this function?

Reply:
The advantage of this function is that if we don't have it, every single place which makes an API calls needs to manually pass in: 
```
 { dataSourceId: dataSourceId };
```
As a query param. If for some reason the implementation code changes in the future or the way we wire this changes, we would need to change all of the instances where API calls are made, vs. just this one function that each API uses to generate that multi-datasource enabled call.

Reply:
@cwperks I am resolving this to make the PR more readable, as it is already cutting off some other comments, feel free to un-resolve if you disagree with that decision.